packages/lab/src/pill/PillNext.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  forwardRef,
+  MouseEvent,
+  PropsWithChildren,
+  HTMLAttributes,
+  KeyboardEvent,
+} from "react";
+import { useWindow } from "@salt-ds/window";
+import { CloseSmallIcon } from "@salt-ds/icons";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { makePrefixer, useButton } from "@salt-ds/core";
+import pillCss from "./PillNext.css";
+import clsx from "clsx";
+
+type ClickEvent = MouseEvent<Element, globalThis.MouseEvent>;
+
+interface PillProps extends HTMLAttributes<HTMLDivElement> {
+  disabled?: boolean;
+  className?: string;
+  icon?: React.ReactNode;
+  onClose?: (e: ClickEvent | KeyboardEvent<HTMLDivElement>) => void;
+}
+
+const withBaseName = makePrefixer("saltPill");
+
+export const Pill = forwardRef<HTMLDivElement, PropsWithChildren<PillProps>>(
+  function Pill(
+    { onClose, onClick, children, className, icon, ...restProps },
+    ref
+  ) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-pill",
+      css: pillCss,
+      window: targetWindow,
+    });
+
+    const clickable = onClick !== undefined;
+    const closable = onClose !== undefined;
+    const interactive = clickable || closable;
+
+    if (interactive) {
+      return (
+        <InteractivePill
+          onClose={onClose}
+          onClick={onClick}
+          className={className}
+          icon={icon}
+          {...restProps}
+        >
+          {children}
+        </InteractivePill>
+      );
+    }
+
+    return (
+      <div ref={ref} {...restProps} className={clsx(withBaseName(), className)}>
+        {icon}
+        <span className={withBaseName("label")}>{children}</span>
+      </div>
+    );
+  }
+);
+
+const InteractivePill = forwardRef<
+  HTMLDivElement,
+  PropsWithChildren<PillProps>
+>(function InteractivePill(
+  { disabled, onClick, className, onClose, children, icon, ...restProps },
+  ref
+) {
+  const clickable = onClick !== undefined;
+  const closable = onClose !== undefined;
+  const { buttonProps, active } = useButton<HTMLDivElement>({
+    disabled: disabled || !clickable,
+    onClick,
+    onKeyUp: clickable
+      ? (e) => {
+          if (e.key === "Enter" || e.key === " ") {
+            onClick?.(e);
+          }
+          restProps.onKeyUp?.(e);
+        }
+      : restProps.onKeyUp,
+    onKeyDown: closable
+      ? (e) => {
+          if (e.key === "Backspace") {
+            onClose(e);
+          }
+          restProps.onKeyDown?.(e);
+        }
+      : restProps.onKeyDown,
+  });
+
+  return (
+    <div
+      ref={ref}
+      role={clickable ? "button" : undefined}
+      {...restProps}
+      {...buttonProps}
+      className={clsx(withBaseName(), className, {
```

Comment:
`className` should be last

Reply:
Order of applied classnames doesn't really matter functionally. So if we are just doing this for readability I think it is easier to keep track of it there rather than as the last param imho. I know this is pure stylistic issue but nonetheless.

```jsx
className={clsx(withBaseName(), {
   [(withBaseName("foo")]: bar
}, className)}
```
vs

```jsx
className={clsx(withBaseName(), className, {
   [(withBaseName("foo")]: bar
})}
```

Reply:
it's because otherwise things can get overridden - cc @Fercas123 who spotted this

Reply:
Interesting, can I see an example @Fercas123?

Reply:
This is more a standard. Most of the other components have it declared this way.